File: LeetCode/Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int index = 0;
        
        for(int i=0; i < t.size(); i++) {
            if (index < s.size() && s[index] == t[i]) index++;
        }
        
        return index == s.size();
    }
};

File: LeetCode/Is Subsequence.ts
function isSubsequence(s: string, t: string): boolean {
  let index = 0;

  for (let i = 0; i < t.length; i++) {
    if (index < s.length && s[index] == t[i]) index++;
  }

  return index == s.length;
}
